src/sentry/static/sentry/app/components/activity/note/input.tsx

```diff
@@ -12,83 +12,66 @@ import marked from 'app/utils/marked';
 import space from 'app/styles/space';
 import textStyles from 'app/styles/text';
 
+import {Mentionable, Mentioned, MentionChangeEvent, CreateError} from './types';
 import Mentionables from './mentionables';
 import mentionStyle from './mentionStyle';
 
-const mentionShape = PropTypes.shape({
-  display: PropTypes.string,
-  email: PropTypes.string,
-  id: PropTypes.string,
-});
-
-class NoteInput extends React.Component {
-  static propTypes = {
-    teams: PropTypes.arrayOf(mentionShape).isRequired,
-    memberList: PropTypes.arrayOf(mentionShape).isRequired,
-
-    // This is the id of the note object from the server
-    // This is to indicate you are editing an existing item
-    modelId: PropTypes.string,
-
-    // The note text itself
-    text: PropTypes.string,
-    error: PropTypes.bool,
-    errorJSON: PropTypes.shape({
-      detail: PropTypes.shape({
-        message: PropTypes.string,
-        code: PropTypes.number,
-        extra: PropTypes.any,
-      }),
-    }),
-    placeholder: PropTypes.string,
-    busy: PropTypes.bool,
-
-    /**
-     * minimum height of the textarea
-     */
-    minHeight: PropTypes.number,
-
-    onEditFinish: PropTypes.func,
-    onUpdate: PropTypes.func,
-    onCreate: PropTypes.func,
-    onChange: PropTypes.func,
-  };
-
-  static defaultProps = {
-    placeholder: t('Add a comment.\nTag users with @, or teams with #'),
-    minHeight: 140,
-    busy: false,
-  };
-
-  constructor(props) {
-    super(props);
+const defaultProps = {
+  placeholder: t('Add a comment.\nTag users with @, or teams with #'),
+  minHeight: 140,
+  busy: false,
+};
 
-    const {text} = props;
-    const defaultText = text || '';
+type Props = {
+  teams: Mentionable[];
+  memberList: Mentionable[];
+  /**
+   * This is the id of the note object from the server
+   * This is to indicate you are editing an existing item
+   */
+  modelId?: string;
+  /**
+   * The note text itself
+   */
+  text?: string;
+  error?: boolean;
+  errorJSON?: CreateError | null;
+  onEditFinish?: () => void;
+  onUpdate?: (data: NoteType) => void;
+  onCreate?: (data: NoteType) => void;
+  onChange?: (e: MentionChangeEvent, extra: {updating?: boolean}) => void;
+} & typeof defaultProps;
+
+type State = {
+  preview: boolean;
+  value: string;
+  memberMentions: Mentioned[];
+  teamMentions: Mentioned[];
+};
 
-    this.state = {
-      preview: false,
-      value: defaultText,
-      memberMentions: [],
-      teamMentions: [],
-    };
-  }
+class NoteInput extends React.Component<Props, State> {
+  state: State = {
+    preview: false,
+    value: this.props.text || '',
+    memberMentions: [],
+    teamMentions: [],
+  };
 
-  cleanMarkdown(text) {
+  cleanMarkdown(text: string) {
     return text
       .replace(/\[sentry\.strip:member\]/g, '@')
       .replace(/\[sentry\.strip:team\]/g, '');
   }
 
-  submitForm = () => {
+  submitForm() {
     if (!!this.props.modelId) {
```

Comment:
as you are touching this file, maybe you could use an early return here avoiding the usage of else 🙃